### ISSUE TITLE
API returns bad data when preserveExistingAuthenticationToken is true

### ISSUE THREAD

**Issue by Serneum** [NemeStats/NemeStats#503]
I've tested this in the API page as well as in code and I've gotten weird data out of it each time.

Request:
```
{
  "userName": "crees",
  "password": "<REDACTED>",
  "preserveExistingAuthenticationToken": true
}
```
Result:
```
{
  "authenticationToken": "ˆ±^Àry6eDÆér\t?kyI´œŠ",
  "authenticationTokenExpirationDateTime": "2016-12-27T05:11:51.053"
}
```

When `preserveExistingAuthenticationToken` is set to `false`, the data seems more sane:
```
{
  "authenticationToken": "491b3f82-8691-4738-8e0e-a23e2faa0b02",
  "authenticationTokenExpirationDateTime": "2016-12-27T05:11:51.053"
}
```

**Comment by jakejgordon** [comment]
You are correct -- there is actually an issue with using preserverExistingAuthenticationToken as the token is encrypted one-way when it gets saved in the database and the API is handing back the encrypted/worthless auth token. I'm going to have to change the API to accept a deviceId so you can have a separate auth token per user per device. This is a more significant change that I probably won't get to for a week or so unfortunately. Thanks for reporting.

**Comment by jakejgordon** [comment]
I'm going to start picking away at this one now. The idea is that in instead of a   "preserveExistingAuthenticationToken" I'll take an optional "deviceId".  This way  when you log in for a given device Id it only resets the auth token for that one device and the other are unaffected.

**Comment by Serneum** [comment]
Would the User-Agent header be something you could use instead? I'm not sure if that would save you any time/work

**Comment by jakejgordon** [comment]
I'm probably going to work on addressing this now. Should have details soon.

**Comment by jakejgordon** [comment]
Resolved in #513
